src/files/add.js

```diff
@@ -1,51 +1,47 @@
 'use strict'
 
-const isStream = require('is-stream')
 const promisify = require('promisify-es6')
-const ProgressStream = require('../utils/progress-stream')
-const converter = require('../utils/converter')
+const ConcatStream = require('concat-stream')
+const once = require('once')
+const isStream = require('is-stream')
+const SendFilesStream = require('../utils/send-files-stream')
 
 module.exports = (send) => {
-  return promisify((files, opts, callback) => {
-    if (typeof opts === 'function') {
-      callback = opts
-      opts = {}
-    }
+  const createAddStream = SendFilesStream(send, 'add')
 
-    opts = opts || {}
+  return promisify((_files, options, _callback) => {
+    if (typeof options === 'function') {
+      _callback = options
+      options = null
+    }
 
-    const ok = Buffer.isBuffer(files) ||
-               isStream.readable(files) ||
-               Array.isArray(files)
+    const callback = once(_callback)
 
-    if (!ok) {
-      return callback(new Error('"files" must be a buffer, readable stream, or array of objects'))
+    if (!options) {
+      options = {}
     }
 
-    const qs = {}
+    const ok = Buffer.isBuffer(_files) ||
+               isStream.readable(_files) ||
+               Array.isArray(_files)
 
-    if (opts['cid-version'] != null) {
-      qs['cid-version'] = opts['cid-version']
-    } else if (opts.cidVersion != null) {
-      qs['cid-version'] = opts.cidVersion
+    if (!ok) {
+      return callback(new Error('"files" must be a buffer, readable stream, or array of objects'))
     }
 
-    if (opts['raw-leaves'] != null) {
-      qs['raw-leaves'] = opts['raw-leaves']
-    } else if (opts.rawLeaves != null) {
-      qs['raw-leaves'] = opts.rawLeaves
-    }
+    const files = [].concat(_files)
 
-    if (opts.hash != null) {
-      qs.hash = opts.hash
-    } else if (opts.hashAlg != null) {
-      qs.hash = opts.hashAlg
-    }
```

Comment:
Where did all of this options went?

Reply:
Here: https://github.com/ipfs/js-ipfs-api/blob/fix/add-stream/src/utils/send-files-stream.js#L87-L103